src/main/kotlin/no/nav/helse/dokument/DokumentService.kt

```diff
@@ -18,19 +20,39 @@ private val dokumentContentTypeCounter = Counter.build()
 
 class DokumentService(
     private val dokumentGateway: DokumentGateway,
+    private val k9MellomlagringGateway: K9MellomlagringGateway,
     private val image2PDFConverter: Image2PDFConverter,
     private val contentTypeService: ContentTypeService
 ) {
 
-    suspend fun hentDokumenter(urls: List<URI>,
-                               aktoerId: AktoerId,
-                               correlationId: CorrelationId): List<Dokument> {
+    suspend fun hentDokumenter(
+        urls: List<URI>,
+        aktoerId: AktoerId?,
+        fodselsnummer: Fodselsnummer,
+        correlationId: CorrelationId
+    ): List<Dokument> {
         logger.trace("Henter ${urls.size} dokumenter.")
-        val alleDokumenter = dokumentGateway.hentDokumenter(
-            urls = urls,
-            aktoerId = aktoerId,
-            correlationId = correlationId
-        )
+
+        val erK9MellomlagringUrl = urls.erK9MellomlagringUrl()
+
+        val alleDokumenter = when (erK9MellomlagringUrl) {
```

Comment:
kan vi ikke da ha 

```
when (aktørId) {
      null -> k9Mellomlagring
      else -> k9Dokument
}
```

Så slipper vi å se på URL'en?

Reply:
Man sender med både aktørId og norskident til k9-joark. Så ingen av de vil være null noen gang. 
Så for å skille mellom dem må man nesten lese url for å forstå hvilken gateway man skal bruke.

Reply:
Ahh sant,
Men motsatt da

```java
      when(identitetsnummer) {
          null -> k9Dokument
          else -> k9Mellomlagring
      }
```

Reply:
Identitetsnummer er heller aldri null.

Reply:
hmm nettopp, får ta frem prosjektet, var litt vrient å skjønne alt ut fra å kun se på diffen

Reply:
Identitetsnummeret er jo søkers fødseslnummer, og den påkrevd for journalføring.
AktørId har også vært påkrevd, da den har vært påkrevd for å hente dokumenter fra k9-dokument.

Da må den gjøres optional, og da ikke bli sendt inn fra prosessering i gcp. Det er egentlig bedre, da det ikke er behov for aktørId da.

Reply:
Nå er jeg med ;) 
Stryk det. Men om ingen kan være null, hvorfor har du da endret AktørId til en nullable?

Reply:
I `JournalforingV1Service` er vel aldri aktørId nullable, så kan aldri bli null inn her heller?

Så når vi fjerner støtte for k9-dokument helt og ikke trenger aktør id så kan egentlig all kode med aktørId fjernes helt

Reply:
Du har helt rett, jeg gjorde den til optional for så å gjøre det du akkurat beskriver her. Må ha sporet av ila. utviklingsprosessen :-P

Reply:
👯

Reply:
Vi trenger jo ikke at aktørId blir sendt inn fra gcp, hvor dens formål har vært å hente dokumenter fra k9-dokument :)